Project/rules.py

```diff
@@ -1,22 +1,72 @@
 import json
+import os
 
 class GameRules:
     """Loads game difficulty settings from JSON."""
-    def __init__(self, difficulty="Normal"):
+    def __init__(self, difficulty="Normal", player_role="Warrior"):
         self.difficulty = difficulty
+        self.player_role = player_role
+        self.filename = "data/game_rules.json"
         self.load_rules()
 
-    def load_rules(self, filename="data/game_rules.json"):
-        """Loads difficulty settings from JSON."""
+    def load_rules(self):
+        """Loads difficulty settings from JSON or creates default settings if missing."""
+        if not os.path.exists(self.filename):
+            self.create_default_rules()  # ✅ Auto-generate game_rules.json if missing
+
         try:
-            with open(filename, "r") as f:
+            with open(self.filename, "r") as f:
                 data = json.load(f)
                 self.__dict__.update(data.get(self.difficulty, {}))
```

Comment:
Using `self.__dict__.update` is risky here since it will overwrite mothods/attributes accidentally. You can use explicit assignments instead if you desire to.

```suggestion
ruleset = data.get(self.difficulty, {})
self.enemy_damage_multiplier = ruleset.get("enemy_damage_multiplier", 1.0)
self.xp_multiplier = ruleset.get("xp_multiplier", 1.0)
self.item_drop_rate = ruleset.get("item_drop_rate", 1.0)
self.starting_gold = ruleset.get("starting_gold", 100)

```